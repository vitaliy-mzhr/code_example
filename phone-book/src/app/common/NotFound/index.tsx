import * as React from 'react';
import { BaseLayout } from '../../layouts';
import NotFoundStyles from './styles';
import { Link } from 'react-router-dom';
import { Button } from '@material-ui/core';

const NotFound = () => (
  <BaseLayout>
    <NotFoundStyles>
      <h2>
        Page not found...
      </h2>
      <Link to={'/'}>
        <Button variant="contained" color="primary">
          Back to main page
        </Button>
      </Link>
    </NotFoundStyles>
  </BaseLayout>
);

export default NotFound;
