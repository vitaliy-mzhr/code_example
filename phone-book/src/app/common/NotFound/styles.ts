import styled from 'styled-components';

const NotFoundStyles = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  
  a {
    text-decoration: none;
  }
`;

export default NotFoundStyles;
