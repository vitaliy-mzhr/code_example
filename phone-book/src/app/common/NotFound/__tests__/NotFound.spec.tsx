import * as React from 'react';
import { shallow } from 'enzyme';
import NotFound from '../../NotFound';

it('NotFound renders without crashing', () => {
  expect(shallow(<NotFound/>)).toMatchSnapshot();
});
