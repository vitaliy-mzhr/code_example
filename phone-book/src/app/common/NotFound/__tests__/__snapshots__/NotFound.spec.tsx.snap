// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound renders without crashing 1`] = `
<BaseLayout>
  <styled.div>
    <h2>
      Page not found...
    </h2>
    <Link
      replace={false}
      to="/"
    >
      <WithStyles(Button)
        color="primary"
        variant="contained"
      >
        Back to main page
      </WithStyles(Button)>
    </Link>
  </styled.div>
</BaseLayout>
`;
